src/Microsoft.AspNet.Razor/SourceLocation.cs

```diff
@@ -16,13 +20,38 @@ public struct SourceLocation : IEquatable<SourceLocation>, IComparable<SourceLoc
         public static readonly SourceLocation Undefined = CreateUndefined();
         public static readonly SourceLocation Zero = new SourceLocation(0, 0, 0);
 
+        /// <summary>
+        /// Initializes a new instance of <see cref="SourceLocation"/>.
+        /// </summary>
+        /// <param name="absoluteIndex">The absolute index.</param>
+        /// <param name="lineIndex">The line index.</param>
+        /// <param name="characterIndex">The character index.</param>
         public SourceLocation(int absoluteIndex, int lineIndex, int characterIndex)
```

Comment:
what scenarios will continue to use this constructor?  for example, could it be `internal` and used only in tests?